src/profile-logic/profile-data.js

```diff
@@ -783,20 +786,45 @@ function _getTimeRangeForThread(
     for (let i = 0; i < markers.length; i++) {
       const startTime = markers.startTime[i];
       const endTime = markers.endTime[i];
+      const markerPhase = markers.phase[i];
       // The resulting range needs to adjust BOTH the start and end of the range, as
       // each marker type could adjust the total range beyond the current bounds.
       // Note the use of Math.min and Math.max are different for the start and end
       // of the markers.
 
       if (startTime !== null) {
```

Comment:
Idea: should we change this code by having a top level switch instead of these 2 if? Wouldn't that be closer of the new logic?
What I'm thinking of:
```js
switch (markerPhase) {
  case INSTANT:
  case INTERVAL_START:
      result.start = Math.min(result.start, startTime);
      result.end = Math.max(result.end, startTime + interval);
      break;
  case INTERVAL_END:
      result.start = Math.min(result.start, endTime);
      result.end = Math.max(result.end, endTime + interval);
      break;
  case INTERVAL:
      result.start = Math.min(result.start, startTime, endTime); // maybe just use startTime
      result.end = Math.max(result.end, startTime + interval, endTime + interval); // maybe just use endTime
      break;
  default:
     throw ...
}
```

Is it possible that `startTime` or `endTime` is null when they shouldn't be (I mean, for example, `startTime` `null` for an `INSTANT` or `INTERVAL_START`)?

What do you think? I believe this is a lot simpler (this brings the simplicity that Markers 2.0 promised) but also has some risk with existing profiles, and would need some more testing.